apps/src/storage/dataBrowser/DataVisualizer.jsx

```diff
@@ -0,0 +1,135 @@
+import React from 'react';
+import Radium from 'radium';
+import PropTypes from 'prop-types';
+import {connect} from 'react-redux';
+import BaseDialog from '@cdo/apps/templates/BaseDialog.jsx';
+import * as dataStyles from './dataStyles';
+import * as rowStyle from '@cdo/apps/applab/designElements/rowStyle';
+
+const INITIAL_STATE = {
+  isVisualizerOpen: false,
+  chartTitle: '',
+  chartType: '',
+  numBins: 0,
+  values: '',
+  xValues: '',
+  yValues: ''
+};
+
+class DataVisualizer extends React.Component {
+  static propTypes = {
+    // from redux state
+    tableColumns: PropTypes.arrayOf(PropTypes.string).isRequired,
+    tableName: PropTypes.string.isRequired,
+    // "if all of the keys are integers, and more than half of the keys between 0 and
+    // the maximum key in the object have non-empty values, then Firebase will render
+    // it as an array."
+    // https://firebase.googleblog.com/2014/04/best-practices-arrays-in-firebase.html
+    tableRecords: PropTypes.oneOfType([PropTypes.object, PropTypes.array])
+      .isRequired
+  };
+
+  state = {...INITIAL_STATE};
+
+  handleClose = () => {
+    this.setState({isVisualizerOpen: false});
+  };
+
+  handleChange = (field, value) => {
+    this.setState({[field]: value});
+  };
+
+  render() {
+    const dropdownField = (fieldName, displayName, values) => (
+      <div id={fieldName} style={rowStyle.container}>
```

Comment:
Can it be implemented without this ID too?

Reply:
Offline: This react component isn't going to be reused, so it doesn't really matter. That being said, Anjali mentioned she's going to get rid of it anyway.